modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -165,6 +165,27 @@ inline void copyFromIE(const IE::Blob::Ptr &blob, MatType &mat) {
     }
 }
 
+// (taken from videoio module)
+static cv::Mutex& getInitializationMutex()
+{
+    static cv::Mutex initializationMutex;
+    return initializationMutex;
+}
+
+// (taken from DNN module)
+IE::Core& getCore(const std::string& id) {
```

Comment:
what is `id` here?

Reply:
IE device name (including hetero plugin), see DNN module for reference code.

Reply:
yep but do we really need its own individual `ie::Core` object for that?

Reply:
There is one know limitation for Myriad device, in example:
```
IECore core;
core.LoadNetwork(net1, "MYRIAD");
core.LoadNetwork(net2, "MYRIAD");
```
works fine, but
```
Core core1, core2;
core1.LoadNetwork(net1, "MYRIAD");
core2.LoadNetwork(net2, "MYRIAD");
```
will fail with NC_ERROR at second LoadNetwork